### PR TITLE
modemmanager improve cleanup

### PR DESCRIPTION
**Pull request by osedl**
Maintainer: @mips171. Please also have a look at this change @aleksander0m
Compile tested: x86_64, lantiq_xrx200 latest openwrt master
Run tested: x86_64, lantiq_xrx200 latest openwrt master, test done = yes

There are situations where the cleanup of interfaces doesn't work as expected. For instance, when ModemManager crashes, cleanup is not triggered. Even if cleanup were to be triggered, the interfaces cannot be queried because the mmcli command is no longer available after a crash. To address this issue, I propose two commits:
1. [ improve cleanup of ifaces](https://github.com/openwrt/packages/commit/0d7a557462702617f73cb17b34aa2e2149ed3151)
2. [move iface cleanup to wrapper script](https://github.com/openwrt/packages/commit/c49aff8595dfaa765f8049dc59bee597eff64d56)

### CODE REVIEW

File: net/modemmanager/files/modemmanager.common [hunk 1]
```diff
@@ -235,24 +235,20 @@ mm_report_modem_wait() {
...
+
+	# Do nothing if there is no sysfs cache
+	[ -f "${MODEMMANAGER_SYSFS_CACHE}" ] || return
+
+	while IFS= read -r sysfs_cache_line; do
+		sysfs_path=$(echo "${sysfs_cache_line}" | awk '{print $1}')
+		status=$(echo "${sysfs_cache_line}" | awk '{print $2}')
+
+		if [ "${status}" = "processed" ]; then
+			mm_log "info" "call cleanup for: ${sysfs_path}"
```
Comment (aleksander0m) [inline net/modemmanager/files/modemmanager.common:248]: I don't think this should be "info" level

File: net/modemmanager/files/usr/sbin/ModemManager-wrapper [hunk 2]
```diff
@@ -28,6 +28,9 @@ main() {
 	mm_report_events_from_cache
 
 	wait "$CHILD"
+
+	# Set all configured interfaces as unavailable
+	mm_cleanup_interfaces
```
Comment (aleksander0m) [inline net/modemmanager/files/usr/sbin/ModemManager-wrapper:33]: There is another `mm_cleanup_interfaces` call just before launching the MM daemon. Was that call doing anything before these changes, given that it required `mmcli` to work?

**Review by aleksander0m — COMMENTED** [review]
I think the changes are reasonable, but @feckert should review and approve.

File: net/modemmanager/files/modemmanager.common (hunk 1 above)
Comment (feckert) [reply net/modemmanager/files/modemmanager.common:248]: Yes this should be a "debug" level

File: net/modemmanager/files/usr/sbin/ModemManager-wrapper (hunk 2 above)
Comment (osedl) [reply net/modemmanager/files/usr/sbin/ModemManager-wrapper:33]: Since the MM daemon should be down at this point, the call likely did nothing before

**Pull request merged**